cmd/titus-storage/emptydir.go

```diff
@@ -0,0 +1,112 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+
+	executorDocker "github.com/Netflix/titus-executor/executor/runtime/docker"
+	"github.com/Netflix/titus-executor/logger"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+func emptyDirRunner(ctx context.Context, command string, config MountConfig) error {
+	switch command {
+	case start:
+		return emptyDirStart(ctx, config)
+	case stop:
+		return emptyDirStop(ctx, config)
+	default:
+		return fmt.Errorf("Command %q unsupported. Must be either start or stop", command)
+	}
+}
+
+func emptyDirStart(ctx context.Context, config MountConfig) error {
+	pod := config.pod
+	nameToVolume := map[string]corev1.Volume{}
+	for _, v := range pod.Spec.Volumes {
+		if v.EmptyDir != nil && v.EmptyDir.SizeLimit != nil && v.Name != "dev-shm" {
+			nameToVolume[v.Name] = v
+		}
+	}
+
+	for _, container := range pod.Spec.Containers {
+		for i := range container.VolumeMounts {
+			vm := container.VolumeMounts[i]
+			vol, ok := nameToVolume[vm.Name]
+			if !ok {
+				continue
+			}
+
+			err := setupEmptyDirMount(ctx, config.taskID, container.Name, &vol, &vm)
+			if err != nil {
+				return err
```

Comment:
Don't think it's really an issue in practice but I wonder if the task runtime dir is cleaned up automatically if task init is canceled due to an error here

Reply:
Yes, it is. That is covered by a failsafe via the systemd unit